### PR TITLE
Add tests for vLLM

### PR DESCRIPTION
**Pull request by sivanantha321**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
- Related #3644 

**Type of changes**
Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Feature/Issue validation/testing**:

Please describe the tests that you ran to verify your changes and relevant result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

- Logs

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Checklist**:

- [x] Have you added unit/e2e tests that prove your fix is effective or that this feature works?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

**Release note**:

```release-note
NONE
```

**Re-running failed tests**

- `/rerun-all` - rerun all failed workflows.
- `/rerun-workflow <workflow name>` - rerun a specific failed workflow. Only one workflow name can be specified. Multiple /rerun-workflow commands are allowed per comment.

### CODE REVIEW

File: python/huggingfaceserver/huggingfaceserver/vllm/vllm_completions.py [hunk 1]
```diff
@@ -343,12 +357,15 @@ def apply_chat_template(
         self,
         messages: Iterable[ChatCompletionRequestMessage,],
     ):
-        return self.tokenizer.apply_chat_template(conversation=messages, tokenize=False)
+        return self.tokenizer.apply_chat_template(
+            conversation=messages, tokenize=False, add_generation_prompt=True
```
Comment (gavrissh) [inline python/huggingfaceserver/huggingfaceserver/vllm/vllm_completions.py]: Ideally this should be a request specific parameter with True as default. For that we need to modify the CreateChatCompletionRequest. [vLLM does that here](https://github.com/vllm-project/vllm/blob/7cd2ebb0251fd1fd0eec5c93dac674603a22eddd/vllm/entrypoints/openai/protocol.py#L177)

File: python/huggingfaceserver/tests/test_vlllm_model.py [hunk 2]
```diff
@@ -0,0 +1,2898 @@
...
+        request = CompletionRequest(request_id=request_id, params=params, context={})
+        response_iterator = await opt_model.create_completion(request)
+        completion = ""
+        log_probs = Logprobs(
+            text_offset=[], token_logprobs=[], tokens=[], top_logprobs=[]
+        )
+        async for resp in response_iterator:
+            assert resp.id == request_id
+            completion += resp.choices[0].text
+            assert len(resp.choices) == 1
```
Comment (Datta0) [inline python/huggingfaceserver/tests/test_vlllm_model.py:1232]: Should we assert on length before doing `completion += resp.choices[0].text`?

File: python/huggingfaceserver/huggingfaceserver/vllm/vllm_completions.py (hunk 1 above)
Comment (sivanantha321) [reply python/huggingfaceserver/huggingfaceserver/vllm/vllm_completions.py]: I agree. We have a separate PR. Let's take this discussion to #3775

File: python/huggingfaceserver/tests/test_vlllm_model.py [hunk 3]
```diff
@@ -0,0 +1,2898 @@
...
+                ",": -1.6155686378479004,
+            },
+            {
+                " fluffy": -5.6248779296875,
+                " a": -1.4977400302886963,
+                " tiny": -3.006150484085083,
+            },
+            {" white": -2.2152767181396484, " fur": -1.9012728929519653},
+        ]
+        assert response.id == expected.id
```
Comment (Datta0) [inline python/huggingfaceserver/tests/test_vlllm_model.py]: Should we have these comparisons as a function and do 
`assert compare_response_to_expected(response,expected)`
```
def compare_response_to_expected(actual,expected, fields_to_compare=[default_fields]) -> bool:
   for field in fields_to_compare:
      if actual.field!=expected.field: 
       return False
  return True
```
I see that we're using this at multiple places

File: python/huggingfaceserver/tests/test_vlllm_model.py (hunk 2 above)
Comment (sivanantha321) [reply python/huggingfaceserver/tests/test_vlllm_model.py:1232]: done

File: python/huggingfaceserver/tests/test_vlllm_model.py (hunk 3 above)
Comment (sivanantha321) [reply python/huggingfaceserver/tests/test_vlllm_model.py]: done it for non stream cases. for stream each case has different scenarios, making it as a common function will be little bit complex.

**Comment by yuzisun** [comment]
Thanks for adding the tests!!

/lgtm
/approve

**Review by yuzisun — APPROVED** [review]

**Pull request merged**